extension/wallet/src/protocol/extract_loan.rs

```diff
@@ -72,23 +83,27 @@ pub async fn extract_loan(
         })
         .unwrap_or_default();
 
-    let loan_txid = borrower.loan_transaction.txid();
+    let timelock = loan_response.timelock;
+    let borrower1 = borrower0
+        .interpret(SECP256K1, loan_response)
+        .map_err(Error::InterpretLoanResponse)?;
+    let loan_txid = borrower1.loan_transaction.txid();
+
     let loan_details = LoanDetails::new(
         btc_asset_id,
-        borrower.collateral_amount,
+        borrower1.collateral_amount,
         collateral_balance,
         usdt_asset_id,
-        borrower.principal_tx_out_amount,
+        borrower1.principal_tx_out_amount,
         principal_balance,
         timelock,
         loan_txid,
-    )
-    .map_err(Error::LoanDetails)?;
+    );
```

Comment:
Looks to me like `LoanDetails` should live in here.

Reply:
used all over the place